### DISCUSSION TITLE
Error : queue limit reached

### DISCUSSION THREAD

**Discussion by rengoku1121** [sidorares/node-mysql2#2136] [Q&A]
I am using mysql2 and createPool connection.
all my query are using db.query('my query');

Is something wrong ? what did I miss ? please help. Thank you

**Comment by sidorares** [comment]
The error is raised when you set `queueLimint` connection config to a value > 0 and the number of queries currently waiting for a free connection is already >= `queueLimint`

https://github.com/sidorares/node-mysql2/blob/01f9b10ba9099538671234bbc7cae883aee34647/lib/pool.js#L78

You need to check your logic and config to see why it happens - it must be a combination of low `queueLimint` value and large number of parallel queries. Example ( promise pool ):

```js
const pool = mysq.createPool({ connectionLimit: 2, queueLimint: 1 });
const results = Promise.all([
  pool.query('select sleep 1'),
  pool.query('select sleep 1'),
  pool.query('select sleep 1'),
]);
```

**Reply by rengoku1121** [reply]
Hello, so far all of my query is using async await type. So how many queueLimit should I set for it to works ?
Thank you :D
